xios_examples/gen_netcdf.py

```diff
@@ -0,0 +1,290 @@
+import sys
+import os
+import argparse
+import netCDF4 as nc
+import numpy as np
+from .dataFunc import dataFunc
+
+# Global defaults here as needed for command line arguments defaults and argument defaults to run function
+defaults = {
+            'func_str': 'sinusiod',
+            'mesh_file': None,
+            'mesh_varname': None,
+            'nlat': 101,
+            'nlon': 100,
+            'nlatr': 81,
+            'nlonr': 80
+           }
+
+def create_ncfile(ncfile, nlat, nlon, func, dim_prefix='', dim_suffix='', data_prefix='', data_suffix=''):
+
+    latname = f'{dim_prefix}latitude{dim_suffix}'
+    lonname = f'{dim_prefix}longitude{dim_suffix}'
+    dataname = f'{data_prefix}data{data_suffix}'
+
+    ncfile.createDimension(latname, nlat)
+    ncfile.createDimension(lonname, nlon)
+    if 'nbounds' not in ncfile.dimensions:
+        ncfile.createDimension('nbounds', 2)
+
+    lat = ncfile.createVariable(latname, np.float32, (latname,))
+    lat.units = 'degrees_north'
+    lat.standard_name = 'latitude'
+    lat.bounds = f'{latname}_bounds'
+
+    step_lat = 180.0/(nlat-1)
+    first_lat = -90.0
+    lat[:] = first_lat + step_lat*np.arange(nlat)
+    lat2d = np.repeat(lat,nlon).reshape(nlat,nlon)
+
+    lat_bnds = ncfile.createVariable(lat.bounds, np.float32, (latname,'nbounds'))
+    lat_bnds[:,0] = lat[:] - step_lat/2.0
+    lat_bnds[:,1] = lat[:] + step_lat/2.0
+
+    lon = ncfile.createVariable(lonname, np.float32, (lonname,))
+    lon.units = 'degrees_east'
+    lon.standard_name = 'longitude'
+    lon.bounds = f'{lonname}_bounds'
+
+    step_lon = 360.0/nlon
+    first_lon = 0.0
+    lon[:] = first_lon + step_lon*np.arange(nlon)
+    lon2d = np.tile(lon,(nlat,1))
+
+    lon_bnds = ncfile.createVariable(lon.bounds, np.float32, (lonname,'nbounds'))
+    lon_bnds[:,0] = lon[:] - step_lon/2.0
+    lon_bnds[:,1] = lon[:] + step_lon/2.0
+
+    data = ncfile.createVariable(dataname, np.float64, (latname,lonname))
+    data.long_name = "input data values"
+    data[:] = func(lat2d, lon2d)
+
+def create_ncfile_unstructured(ncmeshout, meshin_file, meshin_varname, func, add_bounds=True, data_prefix='', data_suffix=''):
+
+    # Create unstructured mesh from lfric mesh netCDF file
```

Comment:
please may we have this as a triple-quote docstring for the function, rather than a hash-comment?

Reply:
done